rest-assured/src/main/java/com/jayway/restassured/internal/http/AuthConfig.java

```diff
@@ -196,23 +196,25 @@ public void oauth2(String accessToken, OAuthSignature signature) {
 
     static class OAuthSigner implements HttpRequestInterceptor {
         protected OAuthConfig oauthConfig;
-        protected OAuth1AccessToken token;
+        protected Token token;
         protected OAuth10aService service;
         protected SignatureType type = SignatureType.Header;
         protected OAuthSignature signature;
         protected boolean isOAuth1 = true;
+        protected boolean addEmptyTokenToBaseString;
 
         public OAuthSigner(String consumerKey, String consumerSecret,
-                           String accessToken, String secretToken, OAuthSignature signature) {
+                           String accessToken, String secretToken, OAuthSignature signature, Boolean addEmptyTokenToBaseString) {
 
             this.oauthConfig = new OAuthConfig(consumerKey, consumerSecret,
                     null, getOAuthSigntureType(signature), null, null, null, null, null, null, null);
             this.token = new OAuth1AccessToken(accessToken, secretToken);
             this.signature = signature;
+            this.addEmptyTokenToBaseString = addEmptyTokenToBaseString;
         }
 
         public OAuthSigner(String accessToken, OAuthSignature signature) {
-            this.token = new OAuth1AccessToken(accessToken, "");
+            this.token = new OAuth2AccessToken(accessToken, "");

```

Comment:
Yes it is. Before I upgraded to latest scribe the class was "**_Token_**" so it did not matter if it was Oauth1 or Oauth2 Signer. But when I updated, I updated both with Oauth1AccessToken and later realized that **_OAuthSigner(String accessToken, OAuthSignature signature)_** is being specifically being used for Oauth2